src/python/WMCore/Cache/GenericDataCache.py

```diff
@@ -27,10 +28,16 @@ def isDataExpired(self):
             return True
         return False
     
-    def getData(self):
+    def getData(self, noFail=True):
         if self.isDataExpired():
-            self.data = self.func(**self.kwargs)
-            self.lastUpdate = int(time.time())
+            try:
+                self.data = self.func(**self.kwargs)
+                self.lastUpdate = int(time.time())
+            except Exception as ex:
+                if noFail:
```

Comment:
It seems this `noFail` argument is not used (yet at least). How about we remove it? In the end it could just mask a real problem in case it is misused (as we had in the workqueue <--> policy interface). Otherwise, how about using `logging.exception` instead?